ale_linters/ruby/steep.vim

```diff
@@ -0,0 +1,142 @@
+call ale#Set('ruby_steep_executable', 'steep')
+call ale#Set('ruby_steep_options', '')
+
+" Find the nearest dir containing a Steepfile
+function! ale_linters#ruby#steep#FindRoot(buffer) abort
+    for l:name in ['Steepfile']
+        let l:dir = fnamemodify(
+        \   ale#path#FindNearestFile(a:buffer, l:name),
+        \   ':h'
+        \)
+
+        if l:dir isnot# '.' && isdirectory(l:dir)
+            return l:dir
+        endif
+    endfor
+
+    return ''
+endfunction
+
+" Rename path relative to root
+function! ale_linters#ruby#steep#RelativeToRoot(buffer, path) abort
+    return substitute(a:path, ale_linters#ruby#steep#FindRoot(a:buffer) . '/', '', '')
+endfunction
+
+function! ale_linters#ruby#steep#GetCommand(buffer) abort
+    let l:executable = ale#Var(a:buffer, 'ruby_steep_executable')
+
+    " steep check needs to apply some config from the file path so:
+    " - steep check can't use stdin (no path)
+    " - steep check can't use %t (path outside of project)
+    " => we can only use %s
+
+    " somehow :ALEInfo shows that ALE still appends '< %t' to the command
+    " => luckily steep check ignores stdin
+
+    " somehow steep has a problem with absolute path to file but a path
+    " relative to Steepfile directory works:
+    " see https://github.com/soutaro/steep/pull/975
+    " => change to Steepfile directory and remove leading path
+
+    let l:buffer_filename = fnamemodify(bufname(a:buffer), ':p')
+    let l:buffer_filename = fnameescape(l:buffer_filename)
+
+    return ale#ruby#EscapeExecutable(l:executable, 'steep')
+    \   . ' check '
+    \   . ale#Var(a:buffer, 'ruby_steep_options')
+    \   . ' ''' . ale_linters#ruby#steep#RelativeToRoot(a:buffer, l:buffer_filename) . ''''
+endfunction
+
+function! ale_linters#ruby#steep#GetType(severity) abort
+    if a:severity is? 'warning'
+    \|| a:severity is? 'information'
+    \|| a:severity is? 'hint'
+        return 'W'
+    endif
+
+    return 'E'
+endfunction
+
+" Handle output from steep
+function! ale_linters#ruby#steep#HandleOutput(buffer, lines) abort
```

Comment:
This is a complex handler function. Add a test for this too. There's a directory for testing functions for handling results.

Reply:
Sure, I will!

Reply:
I should finally be able to get back to complete this within a few weeks.

Reply:
"a few weeks" heh. On it.

Reply:
@w0rp I have rebased and added the handler test.